validator/remote/src/main/java/tech/pegasys/teku/validator/remote/apiclient/OkHttpValidatorRestApiClient.java

```diff
@@ -0,0 +1,259 @@
+  }
+
+  private <T> Optional<T> post(
+      final ValidatorApiMethod apiMethod,
+      final Object requestBodyObj,
+      final Class<T> responseClass) {
+    final String requestBody;
+    try {
+      requestBody = jsonProvider.objectToJSON(requestBodyObj);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+
+    final Request request =
+        new Request.Builder()
+            .url(urlBuilder(apiMethod).build())
+            .post(RequestBody.create(requestBody, APPLICATION_JSON))
+            .build();
+
+    return executeCall(request, responseClass);
+  }
+
+  private HttpUrl.Builder urlBuilder(final ValidatorApiMethod apiMethod) {
+    return new HttpUrl.Builder()
+        .scheme(baseEndpoint.scheme())
+        .host(baseEndpoint.host())
+        .port(baseEndpoint.port())
+        .encodedPath(apiMethod.getPath());
+  }
+
+  private <T> Optional<T> executeCall(final Request request, final Class<T> responseClass) {
+    try (final Response response = httpClient.newCall(request).execute()) {
+      LOG.trace("{} {} {}", request.method(), request.url(), response.code());
+
+      switch (response.code()) {
+        case 200:
+          {
+            final String responseBody = response.body().string();
+            if (responseClass != null) {
+              final T responseObj = jsonProvider.jsonToObject(responseBody, responseClass);
+              return Optional.of(responseObj);
+            } else {
+              return Optional.empty();
+            }
+          }
+        case 202:
+        case 204:
+        case 404:
+        case 503:
+          {
+            return Optional.empty();
+          }
+        case 400:
+          {
+            LOG.warn("Invalid params response from Beacon Node API - {}", response.body().string());
+            return Optional.empty();
+          }
+        default:
+          {
+            final String responseBody = response.body().string();
+            LOG.error(
+                "Unexpected error calling Beacon Node API (status = {}, response = {})",
+                response.code(),
+                responseBody);
+            throw new RuntimeException(
+                "Unexpected response from Beacon Node API (status = "
+                    + response.code()
+                    + ", response = "
+                    + responseBody
+                    + ")");
+          }
+      }
+    } catch (IOException e) {
+      LOG.error("Error communicating with Beacon Node API", e);
+      throw new RuntimeException(e);
+    }
+  }
+
+  private String encodeQueryParam(final Object value) {
+    try {
+      return removeQuotesIfPresent(jsonProvider.objectToJSON(value));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(
+          "Can't encode param of type " + value.getClass().getSimpleName(), e);
+    }
+  }
+
+  private String removeQuotesIfPresent(final String value) {
+    StringUtils.strip(value, "\"");
```

Comment:
We don't seem to use the result of this.

Reply:
Ah, I was playing around with it :)